src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
```

Comment:
we should not have share-tracking specific logic inside viewer-impl.
let's try to make it generic, for example: `updateFragment()`. and move share-tracking specific logic back to `amp-share-tracking.js`